### ISSUE TITLE
Passing in non empty string in query returns: TypeError decode() argument 1 must be string, not None

### ISSUE THREAD

**Issue by njamaleddine** [soundcloud/soundcloud-python#58]
It appears that passing in a non empty string as a query parameter to search for tracks no longer works.

**Ex from the soundcloud api documentation:**
```
import soundcloud

# create a client object with your app credentials
client = soundcloud.Client(client_id='YOUR_CLIENT_ID')

# find all sounds of buskers licensed under 'creative commons share alike'
tracks = client.get('/tracks', q='buskers', license='cc-by-sa')
```

**Stack trace:**
```
TypeError                                 Traceback (most recent call last)
<ipython-input-7-23b0dc747356> in <module>()
----> 1 tracks = client.get('/tracks', q='buskers', license='cc-by-sa')

/Users/nabil/.virtualenvs/soundcloud_test/lib/python2.7/site-packages/soundcloud/client.pyc in _request(self, method, resource, **kwargs)
    128             'proxies': self.options.get('proxies', None)
    129         })
--> 130         return wrapped_resource(make_request(method, url, kwargs))
    131 
    132     def __getattr__(self, name, **kwargs):

/Users/nabil/.virtualenvs/soundcloud_test/lib/python2.7/site-packages/soundcloud/resource.pyc in wrapped_resource(response)
     50     """
     51     # decode response text
---> 52     response_content = response.content.decode(response.encoding)
     53 
     54     try:

TypeError: decode() argument 1 must be string, not None
```

I've tried to change my Client id and use a different account with no success. Passing in an empty string appears to still work though. 
```
tracks = client.get('/tracks', q='', limit=1)
In [13]: print tracks[0].title
Gino Goes To "Entourage" (6/4/15)
```

**Comment by ethanopp** [comment]
I was having the same issue. Changed the query parameter to take a list of strings as opposed to just a single string and it seems to be working. So for your code, try:

tracks = client.get('/tracks', q=[''], limit=1)

Not exactly sure why this is working but it is...

**Comment by njamaleddine** [comment]
Anyone have any luck with this one?

**Comment by ethanopp** [comment]
Thanks for that post njamaleddine.

Made the update and its working now.

**Comment by Wkryst** [comment]
ethanopp did you fixed the issues? It is still not working...

**Comment by nemovimies** [comment]
Not working for me neither :/

**Comment by jpae** [comment]
@ethanopp Would you mind posting a sample URL? I'm getting a 400 Client Error when I pass in a list instead of a single string. I'm getting url: https://api.soundcloud.com/tracks.json?limit=1&client_id=[MY_CLIENT_ID]&q%5B%5D=Country

**Comment by jpae** [comment]
This issue is fixed on version 0.4.2. You no longer need to pass in a list you can pass in a single string and you won't get the type None error.
